utilities/transactions/transaction_lock_mgr.cc

```diff
@@ -646,6 +667,10 @@ TransactionLockMgr::LockStatusData TransactionLockMgr::GetLockStatusData() {
 
   return data;
 }
+std::vector<std::vector<DeadlockInfo>>
+TransactionLockMgr::GetDeadlockInfoBuffer() {
+  return dlock_buffer_.prepareBuffer();
```

Comment:
The read here needs to be done under a lock. Otherwise, you may hit race conditions if you have other threads writing into the vector.